src/main/java/com/redhat/pantheon/use/DateFormatter.java

```diff
@@ -0,0 +1,25 @@
+package com.redhat.pantheon.use;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.Model;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+/**
+ * Created by ben on 4/1/19.
+ * http://blogs.adobe.com/experiencedelivers/experience-management/htl-date-formatting/
+ * https://sling.apache.org/documentation/bundles/models.html#basic-usage
+ */
+@Model(adaptables = SlingHttpServletRequest.class)
+public class DateFormatter {
+
+    public String value;
+
+    @Inject
+    public DateFormatter(@Named("format") String format, @Named("date") Calendar date) {
+        value = new SimpleDateFormat(format).format(date.getTime());
+    }
+}
```

Comment:
Wow, this I want to hear more about! :smile:

Reply:
What do you want to know? :)

This enables the passing of parameters via the data-sly-use property on src/main/resources/apps/pantheon/modules/all/html.html:29.

One note... Injection via constructor annotations works. Injection via field annotations, followed by a @PostConstruct method annotation, did not. No idea why. All I can say is that my favorite phrase "too much magic" applies here.

Reply:
IMO, sorting is a good candidate for being done client-side. The server doesn't have any info the client doesn't have and waiting for the round-trip takes time. That said, I could imagine at least one way to accomplish a server-side sort using URL params. (It's clumsy, and I'm sure there are better ways, but I don't know them offhand).

Reply:
You know, when I had been thinking about sorting, it was sorting titles - and I had thought it was reasonable to always send the full list to the client. Now I realize we're talking about sorting modules, which could be much more numerous. In that case perhaps server-side sorting is the best approach.